hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -234,6 +236,23 @@ private static void checkIfRequiresMultiAssignment(final Collection<DeploymentRe
         }
     }
 
+    private void validateOnlineAssignment(final List<DeploymentRequest> deploymentRequests) {
+        // remove bypassing the weight enforcement as soon as weight can be set
+        // via UI
+        final boolean bypassWeightEnforcement = true;
+        final long assignmentsWithWeight = deploymentRequests.stream()
+                .filter(request -> request.getTargetWithActionType().getWeight() != null).count();
+        final boolean containsAssignmentWithWeight = assignmentsWithWeight > 0;
+        final boolean containsAssignmentWithoutWeight = assignmentsWithWeight < deploymentRequests.size();
```

Comment:
The above suggested change would work for the case when the list of Doploymentrequest has either complete list of weights or no-weights at all. But in the case where there is a mix of both, **containsAssignmentWithWeight** would be true and negation of the same would be false which is logically incorrect. So in all scenarios it would be better to evaluate based on **assignmentsWithWeight** just to make sure we are covering all the combinations of weights:no-weights.